docs/manual-EN/overview/design-and-arc.md

```diff
@@ -0,0 +1,120 @@
+
+   Meta service manages the servers and partitions in the cluster, e.g. records location of the partitions, receives heartbeat from servers, etc. It balances the partitions and manages the communication traffic in case of server failure.
+- Graph space management
+
+   **Nebula Graph** supports multiple graph spaces. Data in different graph spaces are physically isolated. Meta service stores the metadata of all spaces in the cluster and tracks changes that take place in these spaces, like adding, dropping space, modifying graph space configuration (Raft copies).
+-   Schema management
+
+    **Nebula Graph** is a strong typed database.
+
+    - Types of tag and edge properties are recorded by meta service. Supported data types are: int, double, timestamp, list, etc.
+    - Multi-version management, supporting adding, modifying and deleting schema, and recording its version.
+    - TTL (time-to-live) management, supporting automatic data deletion and space reclamation.
+The meta service is stateful, and just like the storage service, it persists data to a key-value store.
+
+
+### Design-thinking: Query Engine
+
+Nebula Graph's query language **nGQL** is a SQL-like descriptive language rather than an imperative one. It's compossible but not embeddable, it uses Shell pipe as an alternative, aka output in the former query acts as the input in the latter one. Key features of nGQL are as follows:
+
+- Main algorithms are built in the query engine
+- Duplicate queries can be avoided by supporting user-defined function (UDF)
+- Programmable
+
+The binary of the query engine is **nebula-graphd**. Each nebula-graphd instance is stateless and never talks to other nebula-graphd. nebula-graphd only talks to the storage service and the meta service. That makes it trivial to expand or shrink the query engine cluster.
+
+ The query engine accepts the message from the client and generates the execution plan after the lexical parsing (Lexer), semantic analysis (Parser) and the query optimization. Then the execution plan will be passed to the execution engine. The query execution engine takes the query plans and interacts with meta server and the storage engine to retrieve the schema and data.
+
+ The main optimizations of the query engine are:
+
+ - Asynchronous and parallel execution
+
+    I/O operations and network transmission are time-consuming. Thus asynchronous and parallel operations are widely adopted in the query engine to reduce the latency and to improve the overall throughput.Also, a separate resource pool is set for each query to avoid the long tail effect of those time-consuming queries.
+
+ - Pushing down computation
+
+    In a distributed system, transferring large amount of data on the network really hurts the overall latency. In **Nebula Graph**, the query engine will make decisions to push some filter and aggregation down to the storage service. The purpose is to reduce the amount of data passing back from the storage.
```

Comment:
"really hurts the overall latency" a little bit weird

Reply:
How about hurts => extend?